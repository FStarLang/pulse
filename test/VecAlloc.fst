module VecAlloc

#lang-pulse

open Pulse
module V = Pulse.Lib.Vec

fn hf ()
{
  let v = V.alloc 1ul 100sz;
  V.free v;
  ()
}
