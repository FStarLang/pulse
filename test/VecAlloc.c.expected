/* krml header omitted for test repeatability */


#include "VecAlloc.h"

void VecAlloc_hf(void)
{
  uint32_t *buf = KRML_HOST_MALLOC(sizeof (uint32_t) * (size_t)100U);
  if (buf != NULL)
    for (uint32_t _i = 0U; _i < (size_t)100U; ++_i)
      buf[_i] = 1U;
  KRML_HOST_FREE(buf);
}

